tests/models/test_base_model.py

```diff
@@ -31,4 +34,36 @@ def test_base_model_raises_error_if_classification_labels_not_provided():
         _CustomModel("classification")
 
 
-# @TODO: add tests for model download
+# Tests for model download
+def test_model_download(request):
+    model: BaseModel = request.getfixturevalue("german_credit_model")
+    project_key = str(uuid.uuid4())
+
+    with utils.MockedProjectCacheDir(project_key):
+        with utils.MockedClient(mock_all=False) as (client, mr):
+            # The model needs to request files
+            utils.register_uri_for_model_meta_info(mr, model, project_key)
+            utils.register_uri_for_model_artifact_info(mr, model, project_key, register_file_contents=True)
```

Comment:
Should we make sure mocked_request are called, and with good params ?

Reply:
Done in #1535